### ISSUE TITLE
EPROTO: protocol error, symlink '../jsesc/bin/jsesc'

### ISSUE THREAD

**Issue by tienqn** [laravel/homestead#1409]
Hi every one, i have a problem when try to `npm install` in laravel project
Hope some can help me. Thank you so much.
## Versions
  - Vagrant: 2.2.7
  - Provider: Virtualbox
  - Homestead: You should be on the most recent `release` branch.

## Host operating system
windows 10

## Homestead.yaml
```yaml
---
ip: "192.168.10.10"
memory: 2048
cpus: 2
provider: virtualbox

authorize: c:/Users/admin/.ssh/id_rsa.pub

keys:
    - c:/Users/admin/.ssh/id_rsa

folders:
    - map: c:/Users/admin/Projects2019/vy_spa/html/labeauty
      to: /home/vagrant/Projects2019/vy_spa/html/labeauty

    - map: c:/Users/admin/Projects2019/wit/admin/web-wit
      to: /home/vagrant/Projects2019/wit/admin/web-wit

    - map: c:/Users/admin/Projects2019/demo/homstead
      to: /home/vagrant/Projects2019/demo/homstead

      type: "nfs"

sites:
    - map: labeauty.test
      to: /home/vagrant/Projects2019/vy_spa/html/labeauty/public

    - map: wit.test
      to: /home/vagrant/Projects2019/wit/admin/web-wit/public

    - map: demo.test
      to: /home/vagrant/Projects2019/demo/homstead

databases:
    - labeauty
    - db_test

features:
    - mariadb: false
    - ohmyzsh: false
    - webdriver: false

```

## Steps to reproduce
1. vagrant up && vagrant ssh
2. cd Projects2019/vy_spa/html/labeauty
3. npm install

![Screenshot_1](https://user-images.githubusercontent.com/54127625/79945083-47ef1380-8497-11ea-914a-1dcd98c883a4.png)

**Comment by svpernova09** [comment]
Our general suggestion for this is to check out https://laravel.com/docs/7.x/homestead#provider-specific-virtualbox or use NPM outside of the VM (your host has more CPU and RAM anyway)

**Comment by purnesh42H** [comment]
The error sometimes return after you shutdown the VM and restart. Below is a fix with repro

Obviously have this in your vagrant file, as suggested above

config.vm.provider "virtualbox" do |v|
    v.customize ["setextradata", :id, "VBoxInternal2/SharedFoldersEnableSymlinksCreate/Folder_Name", "1"]
end

I recommended comment out any sync folder and let default ./(at host) synced at /vagrant(at guest). This will make sure your **shared folder(Folder_Name)** name is always **vagrant** as mentioned here https://www.vagrantup.com/docs/synced-folders/. Also, you won't face issues with ssh

If the symlink protocol error returns after restart or still persists run 

VBoxManage setextradata "VM_Name" VBoxInternal2/SharedFoldersEnableSymlinksCreate/Folder_Name 1

vagrant halt
vagrant up

vagrant ssh

Usually, VBoxManagee is located at C:\Program Files\Oracle\VirtualBox\

Verify working of symlink by going to /vagrant and running

touch test.txt && ln -s test.txt test1.txt

PS: I have tried this on Windows Home which has limited accessibility to security settings

**Comment by vibonacci** [comment]
Even though almost all other sources say adding 
```
config.vm.provider "virtualbox" do |v|
   v.customize ["setextradata", :id, "VBoxInternal2/SharedFoldersEnableSymlinksCreate/Folder_Name", "1"]
end
```
to Vagrantfile is sufficient, I could not make it work without running
```
VBoxManage setextradata "VM_Name" VBoxInternal2/SharedFoldersEnableSymlinksCreate/Folder_Name 1
```
using VBoxManage.exe. So make sure you try that and your VM_Name and Folder_Name are correct.

**Comment by tymoshenkoas** [comment]
Hi, guys! 
I have an error with symlinks, did what @purnesh42H and @vibonacci suggested:
1) I added to Vagrantfile
```
config.vm.provider "virtualbox" do |v|
   v.customize ["setextradata", :id, "VBoxInternal2/SharedFoldersEnableSymlinksCreate/vagrant", "1"]
end
```

"vagrant" is my folder name

2) and ran in my VirtualBox directory
```
VBoxManage setextradata "homestead" VBoxInternal2/SharedFoldersEnableSymlinksCreate/vagrant 1
```

Then I restarted vagrant with --provision and tried to do the test:
```
touch test.txt && ln -s test.txt test1.txt
```

Well, it works in /home/vagrant but doesn't work in /home/vagrant/code/

Also I checked my Windows user permissions and the user can create symlinks.

What could I miss?

P.S. Also I tied .../SharedFoldersEnableSymlinksCreate/vagrant/*. I thought I could help but it didn't.
